Add simple Cilium Envoy Config CRD example

jrajahalme opened the pull request:
- Small fixes to INSTALLATION.md
- Add a Cilium Envoy Config CRD example with instructions

lizrice commented on INSTALLATION.md:
````diff
@@ -44,12 +44,14 @@ In order to use the `hubble` command to observe flows you will want to use port-
 
 ```
-hubble observe --server localhost:4245
+hubble observe -f
````
I think if you're using port-forward, you have to specify the server address (unless you're on the same machine in which case the default setting of the hubble socket works, but in that case I don't think the hubble port-forwarding would be necessary?)

jrajahalme replied:
Port forward forwards a remote (or VM/container) port into a localhost port, and as long as the local port is the hubble default (4245) you do not need to specify the server when using port-forward.

lizrice replied:
Aha, the [default server changed ](https://github.com/cilium/hubble/commit/a1957ac11b09dd3bfd525ab3f8300df5623d3352)in 0.9.0 and I still had 0.8.0. You are correct!

lizrice commented on ENVOY-LISTENER-EXAMPLE.md:
```diff
@@ -0,0 +1,167 @@
+`type.googleapis.com/envoy.config.route.v3.RouteConfiguration`, are
+always available.
+
+Cilium nodes deploy an Envoy image to support Cilium HTTP policy
+enforcement and observability. This build of Envoy has been optimized
+for the needs of the Cilium Agent and does not contain many of the
+Envoy extensions available in the Envoy code base.
+
+To see which Envoy extensions are available, please have a look at
+[this Envoy extensions configuration
+file](https://github.com/cilium/proxy/blob/master/envoy_build_config/extensions_build_config.bzl).
+Only the extensions that have not been commented out with `#` are
+built in to the Cilium Envoy image. Currently this contains the
+following extensions:
+
+- `envoy.filters.http.router`
+- `envoy.filters.listener.tls_inspector`
+- `envoy.filters.network.http_connection_manager`
+- `envoy.filters.network.mongo_proxy`
+- `envoy.filters.network.mysql_proxy`
+- `envoy.filters.network.tcp_proxy`
+- `envoy.stat_sinks.metrics_service`
+- `envoy.transport_sockets.raw_buffer`
+
+We will evolve the list of built in extensions based on customer
+feedback for those extensions to be available to be used from CEC
+CRDs.
```
This is really useful - I'm thinking we should turn this into a Discussion for getting feedback

jrajahalme commented:
Renamed the `examples` directory as `custom-envoy-listener` and moved instructions to there as well (`README.md`). Also test drove the script in a local VM with Kind.

Marked as ready for review :-)
